pkg/storagecapabilities/storagecapabilities.go

```diff
@@ -308,9 +310,17 @@ var storageClassToProvisionerKeyMapper = map[string]func(sc *storagev1.StorageCl
 	},
 	"pxd.portworx.com": func(sc *storagev1.StorageClass) string {
 		// https://docs.portworx.com/portworx-enterprise/operations/operate-kubernetes/storage-operations/manage-kubevirt-vms.html#create-a-storageclass
-		opts := strings.Split(sc.Parameters["sharedv4_mount_options"], ",")
-		if slices.Contains(opts, "vers=3.0") && slices.Contains(opts, "nolock") {
-			return "pxd.portworx.com/nfs"
+		if val, exists := sc.Parameters["backend"]; exists {
+			if val == "pure_block" {
+				return "pxd.portworx.com/pure_block"
+			} else if val == "pure_fa_file" {
+				return "pxd.portworx.com/pure_fa_file"
+			}
+		} else {
```

Comment:
Are you sure about the `else` here? afaiu `sharedv4_mount_options` should be checked even if there is a `backend` parameter (other than `pure_block`/`pure_fa_file`).